monai/handlers/checkpoint_saver.py

```diff
@@ -0,0 +1,194 @@
+        iteration_save_interval (int): save checkpoint or session every N iterations.
+        iteration_n_saved (int): save latest N iterations' checkpoints or sessions, 'None' is to save all.
+        convert_to_single_gpu (bool): if current model is trained by multi GPU and
+            want to save to single GPU model, need to save `object.module` instead.
+
+    Note:
+        CheckpointHandler can be used during training, validation or evaluation.
+        example of saved files:
+            checkpoint_iteration=400.pth
+            checkpoint_iteration=800.pth
+            checkpoint_epoch=1.pth
+            checkpoint_final_iteration=1000.pth
+            checkpoint_key_metric=0.9387.pth
+
+    """
+
+    def __init__(
+        self,
+        save_dir,
+        save_dict,
+        name=None,
+        file_prefix="",
+        save_final=False,
+        save_key_metric=False,
+        key_metric_name=None,
+        key_metric_n_saved=1,
+        epoch_save_interval=0,
+        epoch_n_saved=None,
+        iteration_save_interval=0,
+        iteration_n_saved=None,
+        convert_to_single_gpu=False,
+    ):
+        assert save_dir is not None, "must provide directory to save the checkpoints."
+        self.save_dir = save_dir
+        assert save_dict is not None and len(save_dict) > 0, "must provide source objects to save."
+        if convert_to_single_gpu:
+            for k, v in save_dict.items():
+                if hasattr(v, "module"):
+                    save_dict[k] = v.module
+        self.save_dict = save_dict
+        self.logger = None if name is None else logging.getLogger(name)
+        self.save_final = save_final
+        self.save_key_metric = save_key_metric
+        self.key_metric_name = key_metric_name
+        self.key_metric_n_saved = key_metric_n_saved
+        self.epoch_save_interval = epoch_save_interval
+        self.epoch_n_saved = epoch_n_saved
+        self.iteration_save_interval = iteration_save_interval
+        self.iteration_n_saved = iteration_n_saved
+
+        if self.save_final is True:
+
+            def _final_func(engine):
+                return engine.state.iteration
+
+            self.final_checkpoint = ModelCheckpoint(
+                self.save_dir,
+                file_prefix,
+                score_function=_final_func,
+                score_name="final_iteration",
+                require_empty=False,
+            )
+        if self.save_key_metric is True:
+
+            def _score_func(engine):
+                if isinstance(self.key_metric_name, str):
+                    metric_name = self.key_metric_name
+                elif hasattr(engine.state, "key_metric_name") and isinstance(engine.state.key_metric_name, str):
+                    metric_name = engine.state.key_metric_name
+                else:
+                    raise ValueError("must provde key_metric_name to save best validation model.")
+                return round(engine.state.metrics[metric_name], 4)
```

Comment:
why rounding here? might be a surprise for the users...

Reply:
The original number is very very long, the saved filename is also very long.

Reply:
i see, we might need a comment in the docstring...

Reply:
added in the latest patch.
Thanks.